### PR TITLE
[2.19.x] DDF-UI-199 Added error messaging for west = east

### PR DESCRIPTION
**Pull request by hayleynorton**
#### What does this PR do?
This PR adds an inline error message for when the west coordinate equals the east coordinate in a bounding box geo filter. It also prevents the search from executing and triggers a pop-up error message
#### Who is reviewing it? 
@andrewzimmer @cassandrabailey293 @maryformanek @TeresaHudson @zta6 
#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining 
@shaundmorris 
#### How should this be tested?
Verify you see an inline error message when east = west in all coordinate systems. Verify that drawing a new shape makes the error message disappear in all coordinate systems. Also verify that there's no regression with the south >= north messaging or any other inline error messages. 
#### What are the relevant tickets?
Fixes: codice/ddf-ui#199
#### Screenshots

image
image
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/utils/validation/validation.tsx [hunk 1]
```diff
@@ -270,68 +271,134 @@ function validateLinePolygon(mode: string, currentValue: string) {
...
+        ? 'Left bound cannot equal right bound'
+        : 'West value cannot equal east value',
+    }
+  }
+  return initialErrorState
+}
+
+function validateBoundingBox(key: string, value: any) {
+  const { north, south, west, east } = value.isDms
+    ? getDmsCoords(value)
```
Comment (bdeining) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/utils/validation/validation.tsx:383]: can we split up this triple ternary for readability?

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/utils/validation/validation.tsx [hunk 2]
```diff
@@ -270,68 +271,134 @@ function validateLinePolygon(mode: string, currentValue: string) {
...
+    if (error) {
+      return { error, message }
+    }
+    return validateLatitudes(north, south, isUsngOrUtmUps)
+  } else {
+    const { error, message } = validateLatitudes(north, south, isUsngOrUtmUps)
+    if (error) {
+      return { error, message }
     }
+    return validateLongitudes(west, east, isUsngOrUtmUps)
```
Comment (cassandrabailey293) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/utils/validation/validation.tsx:401]: ❓ why is `validateLongitudes` being called here?

Comment (hayleynorton) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/utils/validation/validation.tsx:401]: If the key is 'lat' but there are no issues with the latitudes, we still want to display an error message if east = west, which is what validateLongitudes checks for

**Comment by shaundmorris** [comment]
build now

**Comment by cassandrabailey293** [comment]
hero successful 🎉 
verified that east = west inline error message appears when east = west. 
verified that the inline error message is removed after drawing a new geometry. 
No regression with the other inline error messages

**Comment by bdeining** [comment]
build now

**Comment by hayleynorton** [comment]
build now

**Pull request merged**